### PR TITLE
Number tas

### PR DESCRIPTION
**Pull request by amyleblond**
Changes:
** Read Me**: added instructions for e2e testing -- found a bug here when I was doing it myself(added card to trello -- currently able to type in a letter instead of number -- total with show 'na' but submit will give server 500 error)
**Models**: course table has 4 new entries for each of the TA positions, added the form.model (read only additions that are new to our form models so far)
**app.js**: added javascript function for totalling number of tas
**home**: updated number on homepage again (LOL)
**number_tas**: entire new template for the page
**urls**: linking number_ta page with views
**views**: logic for saving and updating form dynamically. 

Testing: DB updating
screen shot 2018-01-30 at 7 28 06 pm
totalling & home page button
screen shot 2018-01-30 at 7 28 27 pm
Success message:
screen shot 2018-01-30 at 7 28 45 pm
New course upload:
screen shot 2018-01-30 at 7 31 32 pm
bug: letter is able to be entered but total say n/a
screen shot 2018-01-30 at 7 29 12 pm

### CODE REVIEW

**Review by smwatts — DISMISSED** [review]
https://www.urbandictionary.com/define.php?term=LGTM

Just need to verify with Smucker tomorrow how wants the submit state to appear (saw your card for this in Trello as well). Great job!!

**Comment by amyleblond** [comment]
@smwatts  LOL at the link

File: taworks/taform/models.py [hunk 1]
```diff
@@ -69,6 +73,32 @@ class Meta:
...
+        self.fields['course_name'].disabled
+        self.fields['instructor_name'].widget.attrs['readonly']=True
+        self.fields['instructor_name'].disabled
+        self.fields['instructor_email'].widget.attrs['readonly']=True
+        self.fields['instructor_email'].disabled
+    class Meta:
+        model = Course
+        fields = '__all__'
+        exclude = ('url_hash',)
+        widgets = {'full_ta':Textarea(attrs={'cols':3,'rows':1, 'style':'resize:none;'}),
```
Comment (j2kan) [inline taworks/taform/models.py:97]: https://trello.com/c/cWrlTxjD/68-submitting-letters-in-number-of-tas-page

left a comment there could fix it here or next sprint i'm happy either way

File: taworks/taform/static/taform/js/app.js [hunk 2]
```diff
@@ -16,4 +16,18 @@ function visaFunction() {
...
-}
\ No newline at end of file
+}
+
+function sumTA() {
+	var table = document.getElementById("numberTAs");
+	var row = table.rows.length;
+	var count;
+	for (var i = 1; i < row; i++){
+		var input1 = table.rows[i].cells[5].children[0].value;
```
Comment (j2kan) [inline taworks/taform/static/taform/js/app.js:26]: I understand the cells[5-9] as the columns. 

what does .children[0] do?

File: taworks/taform/templates/taform/number_tas.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+<form>
+    <button class="button" formaction="{% url 'home' %}">Home</button>
+</form>
+
+<head>
+	<title>TA's per Course</title>
+</head>
+<body onload="sumTA()">
+<h2>Assign the Number of TAs Required per Course</h2>
+{% csrf_token %}
```
Comment (j2kan) [inline taworks/taform/templates/taform/number_tas.html]: remove i see you have this below

File: taworks/taform/urls.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
     url(r'static/taform/course_template.csv', views.send_file, name='send_file'),
     url(r'intro.html', views.intro, name='intro'),
     url(r'^instructor/(?P<hash>\w+)/$', views.load_url, name='url'),
+    url(r'number_tas.html',views.assign_tas, name='number_tas'),
```
Comment (j2kan) [inline taworks/taform/urls.py]: space after comma

File: taworks/taform/views.py [hunk 5]
```diff
@@ -273,6 +273,39 @@ def load_url(request, hash):
...
+    ta = False
+    if request.method == 'POST':
+        num = [x for x in models.Course.objects.all()]
+        c_form = models.AssignTA(request.POST)
+        courses = models.Course.objects.all().order_by('section').order_by('course_id').order_by('id')
+        ta = True
+        j = 0
+        for i in courses:
+            obj = models.Course.objects.get(id=i.id)
+            obj.full_ta =c_form.__dict__['data'].getlist('full_ta')[j]
```
Comment (j2kan) [inline taworks/taform/views.py]: space after equals

File: taworks/taform/views.py [hunk 6]
```diff
@@ -273,6 +273,39 @@ def load_url(request, hash):
     url = get_object_or_404(models.Course, url_hash=hash)
     return render_to_response('taform/test.html', {})
 
+def assign_tas(request):
+    if not request.user.is_authenticated:
+        return redirect('login')
+    ta = False
```
Comment (j2kan) [inline taworks/taform/views.py]: can we use a variable name like, is_rankings_submitted ?

**Review by j2kan — DISMISSED** [review]
Left a few comments about a variable name slash spacing after commas.

otherwise LGTM 👍

**Review by j2kan — APPROVED** [review]

**Pull request merged**